app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,87 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+  
+  def show
+    task_id_raw = params[:id]
+    task_id = task_id_raw.to_i
+    @task = Task.find_by(id: task_id)
+    if task_id.to_s != task_id_raw || @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+  
+  def new
+    @task = Task.new
+  end
+  
+  def create    
+    @task = Task.new(task_params) rescue nil
+    if @task
+      begin
+        @task.save!
+      rescue ActiveRecord::RecordInvalid
+        redirect_to new_task_path
+        return
+      end 
+      
+      redirect_to task_path(@task.id)   
+      return  
+    end
+    
+    redirect_to new_task_path
+    return 
```

Comment:
Nice! Your code doesn't quite follow the pattern at Ada that we teach (aka you prefer using `begin ... rescue` and exception handling to handle edge cases), but this code looks fantastic and works super well. I have no complaints-- just be mindful when you're on teams to see what style your team wants to go with.